packages/flutter/lib/src/rendering/object.dart

```diff
@@ -1120,6 +1401,234 @@ abstract class RenderObject extends AbstractNode implements HitTestTarget {
+    assert(() { _debugDoingSemantics = true; return true; });
+    try {
+      _nodesNeedingSemantics.sort((RenderObject a, RenderObject b) => a.depth - b.depth);
+      for (RenderObject node in _nodesNeedingSemantics) {
+        if (node._needsSemanticsUpdate)
+          node._updateSemantics();
+      }
+    } finally {
+      _nodesNeedingSemantics.clear();
+      assert(() { _debugDoingSemantics = false; return true; });
+      Timeline.finishSync();
+    }
+  }
+
+  /// Whether this RenderObject introduces a new box for accessibility purposes.
+  bool get hasSemantics => false;
+
+  /// The bounding box, in the local coordinate system, of this
+  /// object, for accessibility purposes.
+  Rect get semanticBounds;
+
+  bool _needsSemanticsUpdate = true;
+  bool _needsSemanticsUpdateMarkedDescendants = false;
+  SemanticsNode _semantics;
+
+  SemanticsNode get debugSemantics { // only exposed for testing and debugging
+    SemanticsNode result;
+    assert(() {
+      result = _semantics;
+      return true;
+    });
+    return result;
+  }
+
+  /// Mark this node as needing an update to its semantics
+  /// description.
+  ///
+  /// If the change did not only involve a removal or addition of
+  /// semantics, only the change of semantics (e.g. isChecked changing
+  /// from false to true, as opposed to isChecked changing from null
+  /// to true), then you can pass the onlyChanges argument with the
+  /// value true to reduce the cost. If semantics are being added or
+  /// removed, more work needs to be done to update the semantics
+  /// tree. If you pass 'onlyChanges: true' but this node, which
+  /// previously had a SemanticsNode, no longer has one, or previously
+  /// did not set any semantics, but now does, or previously had a
+  /// child that returned annotators, but no longer does, or other
+  /// such combinations, then you will either assert during the
+  /// subsequent call to [flushSemantics()] or you will have
+  /// out-of-date information in the semantics tree.
+  ///
+  /// If the geometry might have changed in any way, then again, more
+  /// work needs to be done to update the semantics tree (to deal with
+  /// clips). You can pass the noGeometry argument to avoid this work
+  /// in the case where only the labels or flags changed. If you pass
+  /// 'noGeometry: true' when the geometry did change, the semantic
+  /// tree will be out of date.
+  void markNeedsSemanticsUpdate({ bool onlyChanges: false, bool noGeometry: false }) {
+    assert(!_debugDoingSemantics);
+    if (!attached || (_needsSemanticsUpdate && onlyChanges &&
+                      (noGeometry || _needsSemanticsUpdateMarkedDescendants))) {
+      return;
+    }
+    if (!noGeometry && (_semantics == null || (_semantics.hasChildren && _semantics.wasAffectedByClip))) {
+      // Since the geometry might have changed, we need to make sure to reapply any clips.
+      visitChildrenForSemantics((RenderObject child) {
```

Comment:
Oh, yeah, that would work. Set a super-dirty bit and then when walking the subtree later, if the bit is set, pass the bit down to make sure the child ignores its dirtyness.